packages/next/src/client/components/react-dev-overlay/pages/ReactDevOverlay.tsx

```diff
@@ -1,66 +1,17 @@
-import * as React from 'react'
+import type { ErrorType } from './OldReactDevOverlay'
+import OldReactDevOverlay from './OldReactDevOverlay'
+import NewReactDevOverlay from '../_experimental/pages/ReactDevOverlay'
 
-import { ShadowPortal } from '../internal/components/ShadowPortal'
-import { BuildError } from '../internal/container/BuildError'
-import { Errors } from '../internal/container/Errors'
-import { ErrorBoundary } from './ErrorBoundary'
-import { Base } from '../internal/styles/Base'
-import { ComponentStyles } from '../internal/styles/ComponentStyles'
-import { CssReset } from '../internal/styles/CssReset'
-import { usePagesReactDevOverlay } from './hooks'
-
-export type ErrorType = 'runtime' | 'build'
-
-interface ReactDevOverlayProps {
+type ReactDevOverlayProps = {
   children?: React.ReactNode
   preventDisplay?: ErrorType[]
   globalOverlay?: boolean
 }
 
-export default function ReactDevOverlay({
-  children,
-  preventDisplay,
-  globalOverlay,
-}: ReactDevOverlayProps) {
-  const {
-    isMounted,
-    displayPrevented,
-    hasBuildError,
-    hasRuntimeErrors,
-    state,
-    onComponentError,
-  } = usePagesReactDevOverlay(preventDisplay)
-
-  return (
-    <>
-      <ErrorBoundary
-        globalOverlay={globalOverlay}
-        isMounted={isMounted}
-        onError={onComponentError}
-      >
-        {children ?? null}
-      </ErrorBoundary>
-      {isMounted ? (
-        <ShadowPortal>
-          <CssReset />
-          <Base />
-          <ComponentStyles />
+export default function ReactDevOverlay(props: ReactDevOverlayProps) {
+  const Overlay = Boolean(process.env.__NEXT_EXPERIMENTAL_NEW_DEV_OVERLAY)
+    ? NewReactDevOverlay
```

Comment:
this will load both resources in dev, we should dynamically load the module based on the flag

Reply:
They'll get bundled anyway dev. Static import is the more robust pattern if tree-shaking is actually applied.